include/fe/fe_abstract.h

```diff
@@ -244,35 +244,68 @@ class FEAbstract : public ReferenceCountedObject<FEAbstract>
   /**
    * \returns The \p xyz spatial locations of the quadrature
    * points on the element.
+   *
+   * It is overwritten by infinite elements since there
+   * \p FEMap cannot be used to compute \p xyz.
    */
+#ifdef LIBMESH_ENABLE_INFINITE_ELEMENTS
+  virtual
+#endif
   const std::vector<Point> & get_xyz() const
   { calculate_map = true; return this->_fe_map->get_xyz(); }
 
+
+#ifdef LIBMESH_ENABLE_INFINITE_ELEMENTS
+  /**
+   * This function is the variant of \p get_JxW() for \p InfFE.
+   * Since J diverges there, a respectize decay-function must be
+   * applied to obtain well-defined quantities.
+   *
+   * For FE, it is equivalent to the common \p get_JxW().
+   */
+  virtual const std::vector<Real> & get_JxWxdecay_sq () const
+  { return get_JxW();}
+#endif
+
   /**
    * \returns The element Jacobian times the quadrature weight for
    * each quadrature point.
+   *
+   * For \p InfFE, use \p get_JxWxdecay_sq() instead.
    */
+#ifdef LIBMESH_ENABLE_INFINITE_ELEMENTS
+  virtual
```

Comment:
IMO it's OK if these functions become `virtual` regardless of whether infinite elements are enabled. There may be other drawbacks but this is not usually called in performance-sensitive parts of the code and it would certainly look a lot cleaner to drop the `ifdef`.

Reply:
I like leaving the ifdef, but I'm happy either way.  Indeed every call to this should either be outside the element loop (where the performance difference is trivial) or at least outside a loop over all test functions on all quadrature points, where the performance difference is nearly trivial.

Reply:
Although... could we turn the zillion ifdef statements into just *one*?  Something like

```
#ifdef LIBMESH_ENABLE_INFINITE_ELEMENTS
#define virtual_for_inffe virtual
#else
#define virtual_for_inffe
#endif
```